lib/postprocessing/image.js

```diff
@@ -261,10 +261,15 @@ async function applyOperations(img, rendition, directories) {
     }
 }
 
-async function applyOutputProperties(img, instructions) {
-    // for reproducible png files
-    if (instructions.fmt === "png") {
+async function applyOutputProperties(img, instructions) {   
+    if (instructions.fmt === "png") { 
```

Comment:
Actually, I would see a better implementation being something similar to:
```
function applyFormatProperties(img, instructions){
    // for reproducible png files
    if (instructions.fmt === "png") {
        img.define("png:exclude-chunks=date");
    }

    if (instructions.fmt === "tif" || instructions.fmt === "tiff") { 
        // to keep same compression as initial file (e.g. for png -> tiff)
        img.define("tiff:preserve-compression=true");
    }

    if (instructions.fmt === "jpg" || instructions.fmt === "jpeg") {
        if(instructions.jpegSize){
            // From http://www.imagemagick.org/Usage/formats/#jpg
            // As of IM v6.5.8-2 you can specify a maximum output filesize for the JPEG image.
            img.define(`jpeg:extent=${instructions.jpegSize}`);
        }
    }
}

async function applyOutputProperties(img, instructions) {
    applyFormatProperties(img, instructions);

    // ... rest of current code

```

Reply:
Yes, but this will just get longer and longer (e.g. we might do more inside the `png` condition), so it's nicer to separate these blocks and not force them "together" via `else if`. Also there are other blocks based on other if conditions.